core/sql/executor/ExExeUtilGet.cpp

```diff
@@ -4940,6 +4946,41 @@ Lng32 ExExeUtilHiveMDaccessTcb::getLengthFromHiveColType(const char* hiveType)
   if ( !strcmp(hiveType, "timestamp"))
     return 26; //Is this internal or display length? REC_DATETIME;
 
+  if ( !strcmp(hiveType, "date"))
+    return 10; //Is this internal or display length? REC_DATETIME;
+  
+  if ( !strncmp(hiveType, "varchar",7) )
+  {
+    //try to get the length
+    char maxLen[32];
+    memset(maxLen, 0, 32);
+    int i=0,j=0;
+    short copyit = 0;
+
+    if(strlen(hiveType) > 39)  return -1;  
+ 
+    for(i = 0; i < strlen(hiveType) ; i++)
+    {
+      if(hiveType[i] == '(')  
+      {
+        copyit=1;
+        continue;
+      }
+      else if(hiveType[i] == ')')  
+        break;
+      if(copyit == 1 )
+      {
+        maxLen[j] = hiveType[i];
+        j++;
+      }
+    }
+
```

Comment:
This code assumes that everything between '(' and ')' is a digit. Is it possible for there to be a non-digit there? (Another way to think about the question: Does the datatype info we are scanning here come from Hive? Or might it come from humans?)